src/optimize.jl

```diff
@@ -135,39 +139,48 @@ function MOI.optimize!(dest::Optimizer{T}, src::MOI.ModelLike) where {T}
     empty!(dest.primal_solution)
     index_map = MOI.copy_to(dest.inner, src)
     ret = _run_minizinc(dest)
-    if isempty(ret)
-        dest.has_solution = false
-    else
-        variable_map = Dict(
-            MOI.get(dest.inner, MOI.VariableName(), x) => x for
-            x in MOI.get(src, MOI.ListOfVariableIndices())
-        )
-        dest.has_solution = true
-        for line in split(ret, "\n")
-            m = match(r"(.+) \= (.+)\;", line)
-            if m !== nothing
-                x = variable_map[m[1]]
-                dest.primal_solution[x] = _parse_result(T, m[2])
+    if !isempty(ret)
+        m_stat = match(r"=====(.+)=====", ret)
+        if m_stat !== nothing
+            @assert length(m_stat.captures) == 1
+            dest.solver_status = m_stat[1]
+        else
+            variable_map = Dict(
+                MOI.get(dest.inner, MOI.VariableName(), x) => x for
+                x in MOI.get(src, MOI.ListOfVariableIndices())
+            )
+            for line in split(ret, "\n")
+                m_var = match(r"(.+) \= (.+)\;", line)
+                if m_var !== nothing
+                    x = variable_map[m_var[1]]
+                    dest.primal_solution[x] = _parse_result(T, m_var[2])
+                end
             end
         end
     end
-    return index_map, false
+    return (index_map, false)
```

Comment:
```suggestion
    return index_map, false
```

Why unnecessary punctuation?